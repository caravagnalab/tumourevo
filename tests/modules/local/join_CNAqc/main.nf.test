nextflow_process {
    name "Test Process JOIN_CNAQC"
    script "modules/local/join_CNAqc/main.nf"
    process "JOIN_CNAQC"

    test("Should run without failures") {
        when {
            params {
                // define module specific parameters here. Example:
                // outdir = "tests/results"
            }            
            process {
                """
                meta = [
                    id : "MSeq_Set06_Set6_42",
                    dataset : "MSeq",
                    patient : "Set06",
                    tumour_sample: "Set6_42",
                    normal_sample: "54_N",
                    cancer_type: "PANCANCER" 
                ]
                rds_list = file(params.local_testdata_base_path + "/from_CNAqc/MSeq_Set06_Set6_42_qc.rds", checkIfExists: true)
                input[0] = tuple(meta, rds_list, meta.tumor_sample)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
