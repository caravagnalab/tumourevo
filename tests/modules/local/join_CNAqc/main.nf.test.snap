{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "normal_sample": "54_N",
                            "cancer_type": "PANCANCER"
                        },
                        "MSeq_Set06_Set6_42_multi_cnaqc.rds:md5,b5521711e9bc816f7bcfa791460dd7a9",
                        null
                    ]
                ],
                "rds": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "normal_sample": "54_N",
                            "cancer_type": "PANCANCER"
                        },
                        "MSeq_Set06_Set6_42_multi_cnaqc.rds:md5,b5521711e9bc816f7bcfa791460dd7a9",
                        null
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-16T12:27:26.021914258"
    }
}